### ISSUE TITLE
Capture a node's IRI in a layout.

### ISSUE THREAD

**Issue by timothee-haudebourg** [spruceid/treeldr#21]
Oftentimes it can be useful to capture the IRI of a node in a layout. For instance, consider the following RDF graph:
```nq
<https://example.com/#a> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/2000/01/rdf-schema#Resource>
_:0 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/2000/01/rdf-schema#Resource> 
```
In Rust, we can define the same graph using the following program:
```rust
/// Node type, representing any `rdfs:Resource` with its id.
struct Node {
  id: Option<IriBuf>
}

let graph = [
  Node {
    id: Some(IriBuf::new("https://example.com/#a").unwrap())
  }
  Node {
    id: None
  }
]
```

We want to be able to define a TreeLDR layout that would translate in Rust into the `Node` struct type. Such a layout would look like this:
```
layout Node for rdfs:Resource {
  id: &rdfs:Resource
}
```
However because the IRI of a node is technically not a property of this node, `id` here is not bound to any property, and TreeLDR has no idea it should hold the actual identifier of the node (if any).

## Solutions

I see two solutions to this issue.

### Define a special property that reflects the identifier of a node.

In this scenario we would define some `tldr:id` property that has a special meaning for the TreeLDR compiler and the different code generators. When a layout field is bound to this property, the processor should know that it correspond to the id of the node in the source graph.

Pros: no need to define new concepts or syntax constructs.
Cons: defines yet another TreeLDR specific term, with a weird semantics.

### Define a special constraint `= &self`.

In this scenario the layout would be defined like this:
```
layout Node for rdfs:Resource {
  id: &rdfs:Resource = &self
}
```

Here `= &self` is a special value constraint that states that `id` should refer to the resource itself.

Pros: no need to define a new property.
Cons: this kind of constraint is not defined in OWL, it would be a TreeLDR only. This may create some new typing complications.

**Comment by timothee-haudebourg** [comment]
I'll go with the special property option.

I defined a `tldr:self` property, of domain and range `rdfs:Resource`. Its special semantics is that the subject and object of this property are equals. It can then be used in a layout to capture the identifier of the considered resource.

```
layout Node for rdfs:Resource {
  tldr:self as id: &rdfs:Resource
}
```
